iroh/src/magicsock.rs

```diff
@@ -979,6 +999,23 @@ impl Handle {
     #[instrument(skip_all)]
     pub(crate) async fn close(&self) {
         trace!(me = ?self.public_key, "magicsock closing...");
+        if self.msock.is_closed() {
+            return;
+        }
+
+        // Only set closing if not already set
+        if self
+            .msock
+            .closing
+            .compare_exchange(false, true, Ordering::Acquire, Ordering::Relaxed)
+            .is_err()
+        {
+            return;
+        }
+
+        // Cancel any running netreports
+        self.shutdown_token_netreport.cancel();
```

Comment:
We now have quite a few variables in play and I'm wondering if we can simplify it: I think we could remove `self.msock.closing` and instead use only two cancel tokens:
`self.shutdown_token_close_start` and `self.shutdown_token_close_endpoint_closed`
The former would be passed to netreport (i.e. replace `shutdown_token_netreport` and the `closing` AtomicBool), and the latter would be used for everything else.

(i.e. instead of having both a cancel token and an atomicbool, just use `cancel_token.is_cancelled()` for when the atomicbool was used)